lib/pbench/server/api/__init__.py

```diff
@@ -147,39 +147,41 @@ def __init__(self):
 
     def post(self):
         global app
-        graphQL = f"{app.config_graphql.get('host')}:{app.config_graphql.get('port')}"
+        graphQL = (
+            f"http://{app.config_graphql.get('host')}:{app.config_graphql.get('port')}"
+        )
         json_data = request.get_json(silent=True)
 
         if not json_data:
-            app.logger.error("GraphQL: Invalid json object %s", request.url)
+            app.logger.warning("GraphQL: Invalid json object {}", request.url)
             abort(400, message="GraphQL: Invalid json object in request")
 
         try:
             # query GraphQL
-            response = requests.post(graphQL, json=json_data)
-
-            # construct response object
-            response = make_response(response.text)
-            response = _corsify_actual_response(response)
-
+            gql_response = requests.post(graphQL, json=json_data)
         except requests.exceptions.ConnectionError:
             app.logger.exception("Connection refused during the GraphQL post request")
-            abort(500, message="Network problem, could not post to GraphQL Endpoint")
+            abort(503, message="Network problem, could not post to GraphQL Endpoint")
         except requests.exceptions.Timeout:
             app.logger.exception("Connection timed out during the GraphQL post request")
             abort(
-                500, message="Connection timed out, could not post to GraphQL Endpoint"
-            )
-        except requests.exceptions.InvalidURL:
-            app.logger.exception("Invalid url during the GraphQL post request")
-            abort(
-                500, message="Invalid GraphQL url, could not complete the post request"
+                504, message="Connection timed out, could not post to GraphQL Endpoint"
```

Comment:
408 would seem better here.

Reply:
Doesn't a `408` mean the _client_ timed out sending a request to the API server, not the API server being a client that timed out sending a request to GraphQL?  It still seems `504` `Gateway Timeout` is the API telling the client that "your request was okay, we received it, but we timed out talking to the GraphQL instance with which we are configured."

Reply:
Ah; actually, 408 means the _server_ timed out waiting for the _client_ ... which I suppose means it got the start of a request but not all of it.

So, yeah, I guess 504 is probably the "nearly bestest" fit.